cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -177,19 +217,21 @@ protected void startUp() throws Exception {
 
       @Override
       public void running() {
-        String announceAddress = configuration.get(Constants.Service.MASTER_SERVICES_ANNOUNCE_ADDRESS,
-                                                   httpService.getBindAddress().getHostName());
-        int announcePort = configuration.getInt(Constants.AppFabric.SERVER_ANNOUNCE_PORT,
-                                                httpService.getBindAddress().getPort());
+        String announceAddress = cConf.get(Constants.Service.MASTER_SERVICES_ANNOUNCE_ADDRESS,
+                                           httpService.getBindAddress().getHostName());
+        int announcePort = cConf.getInt(Constants.AppFabric.SERVER_ANNOUNCE_PORT,
+                                        httpService.getBindAddress().getPort());
 
         final InetSocketAddress socketAddress = new InetSocketAddress(announceAddress, announcePort);
         LOG.info("AppFabric HTTP Service announced at {}", socketAddress);
 
+        // Tag the discoverable's payload to mark it as supporting ssl.
+        byte[] sslPayload = sslEnabled ? Constants.Security.SSL_DISCOVERABLE_KEY.getBytes() : new byte[]{};
```

Comment:
Would we ever check for http? Between https and ssl I would prefer ssl as that is closer to the actual protocol.